src/System.Xml.XmlDocument/tests/XmlAttributeCollectionTests/InsertAfterTests.cs

```diff
@@ -208,5 +208,26 @@ public void InsertAfterReturnsInsertedAttr()
             XmlAttributeCollection target = element.Attributes;
             Assert.Same(newAttr, target.InsertAfter(newAttr, refAttr));
         }
+
+        [Fact]
+        public void InsertAfterRemovesDupRefAttrAtTheEnd()
+        {
+            const string attributeName = "existingAttr";
+            const string attributeUri = "some:existingUri";
+            XmlDocument doc = CreateDocumentWithElement();
+            XmlElement element = doc.DocumentElement;
+            XmlAttribute anotherAttr1 = element.Attributes.Append(doc.CreateAttribute("attr1", "some:uri1"));
+            XmlAttribute anotherAttr2 = element.Attributes.Append(doc.CreateAttribute("attr2", "some:uri2"));
+            XmlAttribute refAttr = element.Attributes.Append(doc.CreateAttribute(attributeName, attributeUri)); //dup
+            XmlAttribute newAttr = doc.CreateAttribute(attributeName, attributeUri);
```

Comment:
Could you also add a test like:
nodeA
nodeB
nodeRef
nodeC

and InsertAfter duplicate of NodeRef after NodeRef? Also please test if inserting dup of nodeRef after nodeC and nodeB if those are not already present